apis/v1beta1/vspherefailuredomain_types.go

```diff
@@ -81,12 +95,66 @@ type Topology struct {
 	// +optional
 	Networks []string `json:"networks,omitempty"`
 
+	// NetworkConfigurations is a list with new network configurations within this failure domain
+	// +optional
+	NetworkConfigurations []NetworkConfiguration `json:"networkConfigs,omitempty"`
+
 	// Datastore is the name or inventory path of the datastore in which the
 	// virtual machine is created/located.
 	// +optional
 	Datastore string `json:"datastore,omitempty"`
 }
 
+// NetworkConfiguration defines a network configuration that should be used when consuming
+// a failure domain.
+type NetworkConfiguration struct {
+	// NetworkName is the network name for this machine's VM.
+	NetworkName string `json:"name,omitempty"`
```

Comment:
This introduces a second place how the NetworkName can be defined (note: there's also `Networks` at the same level as `NetworkConfigurations` now would be.

Reply:
Yes, I added check in webhook that only `Networks` or `NetworkConfigurations` is filled